http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -87,7 +87,7 @@
     NettyNioAsyncHttpClient(DefaultBuilder builder, AttributeMap serviceDefaultsMap) {
         this.configuration = new NettyConfiguration(serviceDefaultsMap);
         this.protocol = serviceDefaultsMap.get(SdkHttpConfigurationOption.PROTOCOL);
-        this.maxStreams = builder.maxHttp2Streams == null ? Integer.MAX_VALUE : builder.maxHttp2Streams;
+        this.maxStreams = builder.maxHttp2Streams == null ? Long.MAX_VALUE : builder.maxHttp2Streams;
```

Comment:
Is this intended to the max value for this setting? If so, we should probably be setting to 2^32 - 1 since the spec defines option values as unsigned 32-bit values: https://httpwg.org/specs/rfc7540.html#SETTINGS

Reply:
Sounds good. Will use `2^32 - 1 `
Yeah, it will use the min value of serverMaxStreams and clientMaxStreams.
```
channel.attr(MAX_CONCURRENT_STREAMS).set(Math.min(clientMaxStreams, serverMaxStreams));
```